### DISCUSSION TITLE
Is there a way to tell a compiler that argument's property is not null if method returns

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/csharplang#6064] [Q&A] [answered]
I have code like this:
```cs
bool CheckProperty(MyType instance) => instance.A?.B?.C != null;

void Main()
{
    if (CheckProperty(instance))
    {
        Console.WriteLine(instance.A.B.C); // <- Warning here
    }
}
```

Is there a way to annotate `CheckProperty` method so that `Console.WriteLine(instance.A.B.C);` line wouldn't show a warning?

**Comment by jnm2** [comment]
No, not currently. The list of possible annotations is at https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis.

**Comment by Joe4evr** [comment, marked as answer]
You can't use the attributes to inform null analysis beyond the scope of the current type.
The best you'll get here today is
```cs
if (instance.A?.B?.C is {} c)
{
     Console.WriteLine(c);
}
```
